pkg/httpclient/client.go

```diff
@@ -0,0 +1,109 @@
+package httpclient
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+)
+
+// Config is used to modify the HTTPClient behaviour when executing requests.
+type Config struct {
+	RetryPolicy  *RetryPolicy
+	CustomClient *http.Client
+}
+
+// HTTPClient abstracts over the std http.Client and provides a retry mechanism.
+type HTTPClient struct {
```

Comment:
I would like to have an interface here, and the struct to be private, so that we can generate mocks (as we do here: [Test](https://github.com/Zondax/golem/blob/161489d19573759377e135eb80d3484b3feb0003/pkg/zrouter/zmiddlewares/zcache_test.go#L20) and [mock](https://github.com/Zondax/golem/blob/main/pkg/zcache/zcache_mock.go)) to use them later in unit tests. 
And perhaps in the interface we can follow the Builder pattern, like resty does, for example: 

```go
resp, err := client.R().
      SetQueryParams(map[string]string{
          "page_no": "1",
          "limit": "20",
          "sort":"name",
          "order": "asc",
          "random":strconv.FormatInt(time.Now().Unix(), 10),
      }).
      SetHeader("Accept", "application/json").
      SetAuthToken("BC594900518B4F7EAC75BD37F019E08FBC594900518B4F7EAC75BD37F019E08F").
      Get("/search_result")
```

Resty: https://github.com/go-resty/resty